packages/react-instantsearch-core/src/lib/useInstantSearchApi.tsx

```diff
@@ -120,6 +141,26 @@ export function useInstantSearchApi<TUiState extends UiState, TRouteState>(
       serverContext.notifyServer({ search });
     }
 
+    if (promiseRef?.current === null && typeof window === 'undefined') {
+      promiseRef.current = wrapPromiseWithState(
+        new Promise((resolve) => {
+          search.once('render', () => {
```

Comment:
what ensures that at this point _all_ widgets have been mounted and not just the first one?

Reply:
something on the TriggerSearch "widget" maybe could be used

Reply:
It's appended after `<InstantSearch>`'s children so they should all have been added at this point, might not with portals or something though